tests/analysis/test_location_identification.py

```diff
@@ -358,6 +366,39 @@ def test_leisure_weighting(self):
         spts.loc[spts["location_id"] == 2, "activity_label"] = "work"
         assert_geodataframe_equal(spts, result)
 
+    def test_only_one_work_location(self):
+        t_work = pd.Timestamp("2021-07-14 18:00:00", tz="utc")
+        h = pd.Timedelta("1h")
+        p = Point(0.0, 0.0)  # not used
+        list_dict = [{"user_id": 0, "location_id": 0, "started_at": t_work, "finished_at": t_work + h, "g": p}]
+        spts = gpd.GeoDataFrame(data=list_dict, geometry="g")
+        spts.index.name = "id"
+        result = osna_method(spts)
+        spts["activity_label"] = "work"
+        assert_geodataframe_equal(result, spts)
+
+    def test_only_one_rest_location(self):
+        t_rest = pd.Timestamp("2021-07-14 07:00:00", tz="utc")
+        h = pd.Timedelta("1h")
+        p = Point(0.0, 0.0)  # not used
+        list_dict = [{"user_id": 0, "location_id": 0, "started_at": t_rest, "finished_at": t_rest + h, "g": p}]
+        spts = gpd.GeoDataFrame(data=list_dict, geometry="g")
+        spts.index.name = "id"
+        result = osna_method(spts)
+        spts["activity_label"] = "home"
+        assert_geodataframe_equal(result, spts)
+
+    def test_only_one_leisure_location(self):
+        t_leis = pd.Timestamp("2021-07-14 01:00:00", tz="utc")
+        h = pd.Timedelta("1h")
+        p = Point(0.0, 0.0)  # not used
+        list_dict = [{"user_id": 0, "location_id": 0, "started_at": t_leis, "finished_at": t_leis + h, "g": p}]
+        spts = gpd.GeoDataFrame(data=list_dict, geometry="g")
+        spts.index.name = "id"
+        result = osna_method(spts)
+        spts["activity_label"] = "home"
```

Comment:
Should this be leisure?
As far as I understood The [osna function](https://github.com/mie-lab/trackintel/blob/72643c60c8f99b85435f6f924df2484ab4255b35/trackintel/analysis/location_identification.py#L330) the outcome should actually be leisure, shouldn't it?

Reply:
[here](https://github.com/mie-lab/trackintel/blob/72643c60c8f99b85435f6f924df2484ab4255b35/trackintel/analysis/location_identification.py#L285) the mean time is calculated but in the case here this should be 01:30:00, which should be classified as leisure. Maybe you can find what is wrong...

Reply:
No, the time is in the timeframe "leisure" but as in the paper suggested everything in leisure should attribute to a "home" label (althougth less weighted than "rest").

Reply:
Ah ok, thanks for the clarification.